samples/net/sockets/tls_configurations/src/main.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(tls_configuration_sample, LOG_LEVEL_INF);
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include <zephyr/posix/sys/eventfd.h>
+
+#include <zephyr/net/socket.h>
+#include <zephyr/net/tls_credentials.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/sys/util.h>
+
+#define CA_CERTIFICATE_TAG 1
+#define PSK_TAG 2
+
+#if defined(CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+static const unsigned char psk[] = { 0x01, 0x02, 0x03, 0x04, 0x05 };
+static const char psk_id[] = "PSK_identity";
+#endif /* CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED */
+
+/* Following certificates (*.inc files) are:
+ * - generated from "create-certs.sh" script
+ * - converted in C array shape in the CMakeList file
+ */
+#if defined(CONFIG_PSA_WANT_ALG_RSA_PKCS1V15_SIGN) || defined(CONFIG_PSA_WANT_ALG_RSA_PSS)
+#define SOME_CERTIFICATE_SUPPORTED
```

Comment:
Maybe rather something like
```suggestion
#define USE_CERTIFICATE
```
?